plugins/plugin-docker/che-plugin-docker-client/src/main/java/org/eclipse/che/plugin/docker/client/params/BuildImageParams.java

```diff
@@ -0,0 +1,111 @@
+/*******************************************************************************
+ * Copyright (c) 2012-2016 Codenvy, S.A.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *   Codenvy, S.A. - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.che.plugin.docker.client.params;
+
+import org.eclipse.che.plugin.docker.client.ProgressMonitor;
+import org.eclipse.che.plugin.docker.client.dto.AuthConfigs;
+
+import java.io.File;
+
+/**
+ * Arguments holder for {@link org.eclipse.che.plugin.docker.client.DockerConnector#buildImage(BuildImageParams, ProgressMonitor)}.
+ *
+ * @author Mykola Morhun
+ */
+public class BuildImageParams {
+
+    private String repository;
+    private AuthConfigs authConfigs;
+    private Boolean doForcePull;
+    private Long memoryLimit;
+    private Long memorySwapLimit;
+    private File[] files;
+
+    /**
+     * @param repository
+     *         full repository name to be applied to newly created image
+     */
+    public BuildImageParams withRepository(String repository) {
+        this.repository = repository;
+        return this;
+    }
+
+    /**
+     * @param authConfigs
+     *         authentication configuration for private registries. Can be null
+     */
+    public BuildImageParams withAuthConfigs(AuthConfigs authConfigs) {
+        this.authConfigs = authConfigs;
+        return this;
+    }
+
+    /**
+     * @param doForcePull
+     *         is pull with force
+     */
+    public BuildImageParams withDoForcePull(boolean doForcePull) {
+        this.doForcePull = doForcePull;
+        return this;
+    }
+
+    /**
+     * @param memoryLimit
+     *         memory limit for build in bytes
+     */
+    public BuildImageParams withMemoryLimit(long memoryLimit) {
+        this.memoryLimit = memoryLimit;
+        return this;
+    }
+
+    /**
+     * @param memorySwapLimit
+     *         total memory in bytes (memory + swap), -1 to enable unlimited swap
+     */
+    public BuildImageParams withMemorySwapLimit(long memorySwapLimit) {
+        this.memorySwapLimit = memorySwapLimit;
+        return this;
+    }
+
+    /**
+     * @param files
+     *         files that are needed for creation docker images (e.g. file of directories used in ADD instruction in Dockerfile).
+     *         One of them must be Dockerfile.
+     */
+    public BuildImageParams withFiles(File... files) {
```

Comment:
I would say we can improve that by addition of syntax sugar methods: addFile(File file), addFiles(File... files)
And even addDockerfile(Dockerfile dockerfile)
WDYT?